src/Features/Core/Portable/DocumentationComments/AbstractDocumentationCommentFormattingService.cs

```diff
@@ -102,10 +103,14 @@ public void AppendSingleSpace()
             public void AppendString(string s)
             {
                 EmitPendingChars();
-
-                Builder.Add(new TaggedText(TextTags.Text, s, Style, NavigationTarget.target, NavigationTarget.hint));
+                Builder.Add(new TaggedText(TextTags.Text, NormalizeLineEndings(s), Style, NavigationTarget.target, NavigationTarget.hint));
 
                 _anyNonWhitespaceSinceLastPara = true;
+
+                // XText.Value returns a string with `\n` as the line endings, causing
+                // the end result to have mixed line-endings. So normalize everything to `\r\n`.
+                // https://www.w3.org/TR/xml/#sec-line-ends
+                static string NormalizeLineEndings(string input) => input.Replace("\n", "\r\n");
```

Comment:
why do we need this though?  what is wrong with the doc comment having \n in it?

Reply:
➡️ This function ensures that non-normalized XML content uses the same end-of-line character as `s_newlinePart`. It could be cleaned up in the future (e.g. if we wanted to use `Environment.Newline` instead), but the approach in this PR keeps the result consistent with current precedent.